### PR TITLE
[new release] xapi-rrd (1.9.0)

### PR DESCRIPTION
**Pull request by psafont**
RRD library for use with xapi

- Project page: <a href="https://github.com/xapi-project/xcp-rrd">https://github.com/xapi-project/xcp-rrd</a>

##### CHANGES:

* ocamlformat: apply new formatting
* unix: remove code from module
* CA-367236: replace ezjsonm with yojson
* Add license to opam metadata and delete deprecated package
* ci: drop travis
* ci: use github
* maintenance: format code with ocamlformat
* chore: prepare for ocamlformat, use dune 2.0

### CODE REVIEW

**Comment by psafont** [comment]
The issues in the CI seem to be with the ppc64 and with dependencies: ppx_deriving_rpc and xapi_stdext_unix (which didn't fail when using 5.0 unlike now)

File: packages/xapi-rrd/xapi-rrd.1.9.0/opam [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+depends: [
+  "ocaml" {>= "4.05"}
+  "dune" {>= "2.0.0"}
+  "base-bigarray"
+  "base-unix"
+  "ppx_deriving_rpc"
```
Comment (mseri) [inline packages/xapi-rrd/xapi-rrd.1.9.0/opam:20]:
```suggestion
  "ppx_deriving_rpc" {>= "6.1.0"}
```
This should fix the ppx issue in 4.07

**Comment by mseri** [comment]
Ocaml 5.0 fails with
```
# (cd _build/default && /home/opam/.opam/5.0/bin/ocamlopt.opt -w -40 -g -o lib_test/unit_tests.exe /home/opam/.opam/5.0/lib/astring/astring.cmxa /home/opam/.opam/5.0/lib/cmdliner/cmdliner.cmxa /home/opam/.opam/5.0/lib/uutf/uutf.cmxa /home/opam/.opam/5.0/lib/alcotest/stdlib_ext/alcotest_stdlib_ext.cmxa /home/opam/.opam/5.0/lib/fmt/fmt.cmxa /home/opam/.opam/5.0/lib/fmt/fmt_cli.cmxa /home/opam/.opam/5.0/lib/re/re.cmxa /home/opam/.opam/5.0/lib/stdlib-shims/stdlib_shims.cmxa /home/opam/.opam/5.0/lib/alcotest/engine/alcotest_engine.cmxa /home/opam/.opam/5.0/lib/ocaml/unix/unix.cmxa /home/opam/.opam/5.0/lib/fmt/fmt_tty.cmxa /home/opam/.opam/5.0/lib/alcotest/alcotest.cmxa -I /home/opam/.opam/5.0/lib/alcotest /home/opam/.opam/5.0/lib/result/result.cmxa /home/opam/.opam/5.0/lib/base64/base64.cmxa /home/opam/.opam/5.0/lib/rresult/rresult.cmxa /home/opam/.opam/5.0/lib/rpclib/core/rpclib_core.cmxa /home/opam/.opam/5.0/lib/yojson/yojson.cmxa /home/opam/.opam/5.0/lib/rpclib/json/json.cmxa /home/opam/.opam/5.0/lib/xmlm/xmlm.cmxa /home/opam/.opam/5.0/lib/rpclib/internals/internals.cmxa /home/opam/.opam/5.0/lib/rpclib/xml/xml.cmxa /home/opam/.opam/5.0/lib/rpclib/rpclib.cmxa lib/rrd.cmxa /home/opam/.opam/5.0/lib/fd-send-recv/fd_send_recv.cmxa -I /home/opam/.opam/5.0/lib/fd-send-recv /home/opam/.opam/5.0/lib/logs/logs.cmxa /home/opam/.opam/5.0/lib/ocaml/threads/threads.cmxa /home/opam/.opam/5.0/lib/sexplib0/sexplib0.cmxa /home/opam/.opam/5.0/lib/ppx_sexp_conv/runtime-lib/ppx_sexp_conv_lib.cmxa /home/opam/.opam/5.0/lib/xapi-backtrace/backtrace.cmxa /home/opam/.opam/5.0/lib/xapi-stdext-pervasives/xapi_stdext_pervasives.cmxa /home/opam/.opam/5.0/lib/xapi-stdext-unix/xapi_stdext_unix.cmxa -I /home/opam/.opam/5.0/lib/xapi-stdext-unix lib_test/.unit_tests.eobjs/native/dune__exe__Unit_tests.cmx)
# /usr/bin/ld: /home/opam/.opam/5.0/lib/fd-send-recv/libfd_send_recv_stubs.a(fd_send_recv_stubs.o): in function `stub_unix_send_fd':
# /home/opam/.opam/5.0/.opam-switch/build/fd-send-recv.2.0.1/_build/default/lib/fd_send_recv_stubs.c:67: undefined reference to `convert_flag_list'
# /usr/bin/ld: /home/opam/.opam/5.0/lib/fd-send-recv/libfd_send_recv_stubs.a(fd_send_recv_stubs.o): in function `stub_unix_recv_fd':
# /home/opam/.opam/5.0/.opam-switch/build/fd-send-recv.2.0.1/_build/default/lib/fd_send_recv_stubs.c:128: undefined reference to `convert_flag_list'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/fd-send-recv.2.0.1/_build/default/lib/fd_send_recv_stubs.c:174: undefined reference to `alloc_small'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/fd-send-recv.2.0.1/_build/default/lib/fd_send_recv_stubs.c:180: undefined reference to `copy_string'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/fd-send-recv.2.0.1/_build/default/lib/fd_send_recv_stubs.c:183: undefined reference to `alloc_small'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/fd-send-recv.2.0.1/_build/default/lib/fd_send_recv_stubs.c:163: undefined reference to `failwith'
# /usr/bin/ld: /home/opam/.opam/5.0/lib/xapi-stdext-unix/xapi_stdext_unix.a(xapi_stdext_unix__Unixext.o): in function `camlXapi_stdext_unix__Unixext__double_fork_1439':
# /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext.ml:572: undefined reference to `unix_exit'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext.ml:571: undefined reference to `unix_exit'
# /usr/bin/ld: /home/opam/.opam/5.0/lib/xapi-stdext-unix/xapi_stdext_unix.a(xapi_stdext_unix__Unixext.o): in function `camlXapi_stdext_unix__Unixext__1':
# :(.data+0x18f8): undefined reference to `unix_exit'
# /usr/bin/ld: /home/opam/.opam/5.0/lib/xapi-stdext-unix/libxapi_stdext_unix_stubs.a(unixext_open_stubs.o): in function `stub_stdext_unix_open_direct':
# /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_open_stubs.c:53: undefined reference to `convert_flag_list'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_open_stubs.c:58: undefined reference to `string_length'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_open_stubs.c:58: undefined reference to `stat_alloc'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_open_stubs.c:61: undefined reference to `enter_blocking_section'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_open_stubs.c:67: undefined reference to `leave_blocking_section'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_open_stubs.c:68: undefined reference to `stat_free'
# /usr/bin/ld: /home/opam/.opam/5.0/lib/xapi-stdext-unix/libxapi_stdext_unix_stubs.a(unixext_write_stubs.o): in function `stub_stdext_unix_write':
# /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_write_stubs.c:49: undefined reference to `enter_blocking_section'
# /usr/bin/ld: /home/opam/.opam/5.0/.opam-switch/build/xapi-stdext-unix.4.19.0/_build/default/lib/xapi-stdext-unix/unixext_write_stubs.c:51: undefined reference to `leave_blocking_section'
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking (exit code 1)
```
which seems to indicate that we need to mark `xapi-stdext` as incompatible with ocaml 5

**Comment by mseri** [comment]
The PPC64 error happens in the crowbar tests and seems to be related to some overflow, do you want to have a look if there is some bug behind or you are fine with it?

**Comment by psafont** [comment]
> do you want to have a look if there is some bug behind or you are fine with it?

I'm fine with it, seeing no other arch is affected, how can I mark the arch incompatibility?

**Comment by mseri** [comment]
`available: arch != "ppc64"` but I would not do, it is only one test failing, it may work fine for most uses

**Pull request merged**